src/app.controller.ts

```diff
@@ -1,11 +1,12 @@
-import { Controller, Get, VERSION_NEUTRAL } from '@nestjs/common';
+import { Controller, Get, Inject, VERSION_NEUTRAL } from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
 import { AppService } from './app.service';
+import { tokens } from './config/constants';
 
 @ApiTags('App')
 @Controller({ version: VERSION_NEUTRAL })
 export class AppController {
-  constructor(private readonly appService: AppService) {}
+  constructor(@Inject(tokens.appService) private appService: AppService) {}
```

Comment:
🎗️ As vitest uses esbuild to build JS, automatic dependency injection is not supported, so we need to manually inject each dependency